openj9Settings.mk

```diff
@@ -65,7 +66,12 @@ ifneq (, $(findstring current, $(JCL_VERSION)))
 			JAVA_SHARED_LIBRARIES_DIR:=$(TEST_JRE_LIB_DIR)$(D)$(ARCH_DIR)$(D)$(VM_SUBDIR)
 			J9VM_PATH=$(TEST_JRE_LIB_DIR)$(D)$(ARCH_DIR)$(D)j9vm
 		endif
-		ADD_JVM_LIB_DIR_TO_LIBPATH:=export LIBPATH=$(Q)$(LIBPATH)$(P)$(TEST_JRE_LIB_DIR)$(D)$(VM_SUBDIR)$(P)$(JAVA_SHARED_LIBRARIES_DIR)$(P)$(TEST_JRE_BIN)$(D)j9vm$(Q);
+		LIBPATH_VAL=$(Q)$(LIBPATH)$(P)$(TEST_JRE_LIB_DIR)$(D)$(VM_SUBDIR)$(P)$(JAVA_SHARED_LIBRARIES_DIR)$(P)$(TEST_JRE_BIN)$(D)j9vm$(Q)
+		CHECK_VM_SUBDIR=
+		ifeq (default, $(VM_SUBDIR))
+			CHECK_VM_SUBDIR=LIBPATH_VAL=$(LIBPATH_VAL); echo $(Q)$(JVM_OPTIONS)$(Q) | grep -q -- '-Xcompressedrefs' && LIBPATH_VAL=$$(echo $(Q)$$LIBPATH_VAL$(Q) | sed 's/default/compressedrefs/g')
```

Comment:
The substitution of default for compressedrefs seems a bit fragile. Would it be better to replace `default:` (or `default;`) on Windows?

Reply:
updated. Please review again. Thanks